fuels-core/src/code_gen/abigen.rs

```diff
@@ -111,18 +111,25 @@ impl Abigen {
         } else {
             (
                 quote! {
-                    use fuels_contract::contract::{Contract, ContractCall, CompiledContract};
+                    use fuels_contract::contract::{Contract, ContractCall};
                     use fuel_gql_client::client::FuelClient;
+                    use fuel_tx::ContractId;
+                    use std::str::FromStr;
                 },
                 quote! {
                     pub struct #name {
-                        compiled: CompiledContract,
+                        contract_id: ContractId,
                         fuel_client: FuelClient
                     }
 
                     impl #name {
-                        pub fn new(compiled: CompiledContract, fuel_client: FuelClient) -> Self {
-                            Self{ compiled, fuel_client }
+                        pub fn new(contract_id: String, fuel_client: FuelClient) -> Self {
+                            let contract_id = ContractId::from_str(&contract_id).unwrap();
+                            Self{ contract_id, fuel_client }
+                        }
+                        pub fn new_default(fuel_client: FuelClient) -> Self {
+                            let contract_id = ContractId::new([0xFFu8;32]);
+                            Self{ contract_id, fuel_client }
```

Comment:
Can we document this? It's not immediately clear, from the name alone, what the differences are between instantiating a contract with `new` and `new_default`.